File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    zerocnt = 0
    cnt = 0
    while len(s)>1 :
        zerocnt+= s.count('0')
        s = s.replace('0','')
        
        c = len(s)
        
        tmp = ''
        while c!=0:
            tmp = str(c%2) + tmp 
            c = c//2
  
        cnt += 1

        s = tmp
      
        

            
    answer.append(cnt)
    answer.append(zerocnt)
        
    
    return answer

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int [2];
        int zerocnt = 0 ;
        int howmanychange= 0 ;
        
        while (true)
        {
            String news = s.replace("0","");
            zerocnt += (s.length()-news.length());
            
            s = news;
            
            howmanychange +=1;
            
            if (s.length() ==1 )
            {
                break;
            } 
            
            //s의 길이를 2진법으로
            StringBuilder sb = new StringBuilder();
            
            int fornews = s.length();
            
            int twozin = 0 ;
            while (fornews > 0)
            {
                int left = fornews%2;
                sb.append(String.valueOf(left));
                fornews = fornews/2;
            }
            
            s= sb.reverse().toString();
            
            
        }
        
        answer[0] = howmanychange;
        answer[1] = zerocnt;
        
        return answer;
    }
}